start_cerberus.py

```diff
@@ -259,6 +261,26 @@ def main(cfg):
                     logging.info("Route url: %s" % route)
                 logging.info("")
 
+            # Aggregate the status and publish it
+            cerberus_status = watch_nodes_status and watch_namespaces_status \
+                and watch_cluster_operators_status and server_status
+
+            if custom_checks:
+                if iteration == 1:
+                    custom_checks_imports = []
+                    for check in custom_checks:
+                        my_check = ".".join(check.replace("/", ".").split(".")[:-1])
+                        my_check_module = importlib.import_module(my_check)
+                        custom_checks_imports.append(my_check_module)
+
+                for check in custom_checks_imports:
+                    check_returns = check.main()
```

Comment:
what if the main() function took forever to return response back ? does cerberus know how to handle that scenario ?. In general if the custom check got crashed for some reason does cerberus taken control over ?

Reply:
Cerberus just calls the custom check functions. It doesn't handle the scenario (take over the control) when when custom check crashes.

Reply:
If the custom check encounters an error/exception, Cerberus handles it and continues with the next iteration. But in scenarios where the control doesn't flow back to main_function in Cerberus (for example, infinite for/while loop), Cerberus doesn't handle this.